### ISSUE TITLE
Improve accessibility

### ISSUE THREAD

**Issue by dstorey** [MicrosoftEdge/Status#36]
We use WAI-ARIA landmark roles and attributes in the index.html template but they’re missing in some of the others, such as features.html

Some of the existing ARIA can also be improved. 
- The feature sections don’t include ARIA. Should use `role="tab list"` and associated attributes. [complete]
- The filter drop downs don’t include ARIA. Not sure if these should be `button` and `listbox`, or `menubar` or something else (investigating)
- Apparently the Live Region doesn’t work, and should only be on the feature count, not the entire feature area too. We should also add `aria-atomic`. [complete]
- We should add a skip link for keyboard users and screen readers. Only Jaws 15 supports native skip link functionality via `main` (which we don't use anyway as we only have 1 `article`.
- We can remove `role="form"` as it is duplicating well supported native semantics [complete]
- The tags added by the filter only have the term as their label. This doesn't indicate to a screen reader that clicking it will remove that term. We should update the label seen by SR (or ignore this if we remove the tags) [complete. via title attribute]
- The menubar doesn't work well with the keyboard and screen readers 

Accessibility audit results:
- Collapse button in accordion doesn't have a label. Should be solved with the aria work.
- `dropdown`, `dropdown-toggle`, `dropdown-menu` and the checkboxes have `onclick` handlers but are not focusable. Likely why the keyboard navigation with Screen Readers doesn't work
- A bunch of elements don’t have strong enough contrast between foreground and background.  We can’t fix this for the most part without changing the design. Elements are:
  - "Platform status" text
  - "beta" text (will go away with final)
  - "features found" and result count
  - The filter drop down labels 
  - The item labels in the drop downs (these are "disabled" so I don't think we can solve these without making them look enabled)
- A number of elements are focusable but obscured:
  - the "platform status" element. This is an artefact of how the original modern.ie menu was built. Should go away with the new menu anyway
  - A few of the spec links and browser icon links. Probably related to how some of the items have odd overlapping text/links. This needs to be fixed for that anyway.

I’ll update this as we find more issues.

**Comment by dstorey** [comment]
Started work on this.

**Comment by dstorey** [comment]
@molant has made the aria state attributes dynamic now, fixed the live regions (hopefully) and removed the role="form"

I’ve added the skeleton aria attributes to the filter drop downs. I’m not 100% sure they're the correct ones, but they should at least work until we know which is the most appropriate. 

@molant: In featuresorting.html : 
- `aria-pressed` on `a role="button"` needs to be true when the drop down is showing, and false when not. ref: https://github.com/InternetExplorer/Status.IE/blob/a11y-improvements/app/templates/featuresorting.html#L3
- The same element needs a unique ID, so that `aria-labelledby` (on the `ul`) can reference it. ref: https://github.com/InternetExplorer/Status.IE/blob/a11y-improvements/app/templates/featuresorting.html#L7
- On the same line, aria-multiselectable should be true for the checkbox drop downs and false for the radio button one (the first drop down)

**Comment by dstorey** [comment]
Now the accordion `<header>` and the `<button>` inside are both focusable and both perform the same task. We should consider removing the button and just adding the styling of the button another way (perhaps with ::after) or giving the button `tabindex="-1"` to remove it from the tab queue. It is overkill to have two tab stops that perform the same action.

Another tab focus issue is that the spec links get focus after the header/button. Not so bad in itself, but those sections suppose to be collapsed, and it visually only displays the links and not the rest of the content. I believe the CSS uses some form of overflow technique to hide the bodies. Now we have aria hooked up we should be able to just use display: none; instead. I'll investigate this.

**Comment by molant** [comment]
@dstorey I've addressed most of your feedback. The only item I think is missing right now is the keyboard accessibility with the menus and the skip link.
All yours!

**Comment by dstorey** [comment]
feedback from Léonie about the main menu

&lt;snip/>
"All that said, if the links are just links (leading to other pages), then a menu probably isn’t the right design pattern to use in the first place. A menu is supposed to emulate the menubar within a software application, so unless you need to display child menus etc. it’s probably best just to remove all the ARIA and leave them as a set of links within a list."

I’ve removed the aria from the main nav at it seems to be causing issues and we’re not using actual drop down menus there. Those roles were left over from the menu I copied from the Modern.ie code, and I don't think they're appropriate. Those changes are on the interop-accessibility branch.

**Comment by dstorey** [comment]
Feedback about the filter drop downs (from the current live site, not the redesign)

"This is more difficult to use with a screen reader. It works up to a point in IE (with Jaws 15), but not at all in Firefox.

In both browsers I can see buttons for “Sorting”, “Technologies”, “Browsers” and “Status”. They’re reported as toggle buttons, because of the aria-pressed attribute present on each one.

In IE, if I activate one of the buttons (for example “Browsers”), it’s “pressed” state is reported, and a dropdown appears beneath it. I can read the different options – “All”, “Internet Explorer”, “Chrome”, “Firefox”, “Safari” and “Opera”, but can’t interact with those options at all.

In Firefox, pressing the button causes the “Browser” button to become an “All” button, with its pressed state indicated. I can tell there is a listbox present, but can’t access the content at all. Activating the “All” button causes it to return to being the “Browser” button.

It’s difficult to say without being familiar with the code, but it seems as though something is blocking the native HTML <input> semantics from being exposed to AT. My hunch is that it’s the JS customisation."

I think a lot of these issues will be fixed in the current interop-accessibility branch? I’ve changed from an anchor to buttons which removed the need to have the tab index and role="button" and fixed a lot of the scrolling when pressing space bar issues.

**Comment by dstorey** [comment]
Closing this. If there are any remaining issues we can open separate bugs.
